.github/workflows/update-pinned-deps.yml

```diff
@@ -0,0 +1,74 @@
+name: Update pinned Python dependencies for the actions
+
+on:
+  push:
+    branches: [main]
+    paths: ['repo/pyproject.toml']
+  schedule:
+    - cron:  '21 9 * * 1'
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  update-dependencies:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # for pushing a branch
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+
+      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+        with:
+          python-version: "3.11"
+
+      - name: Install pip-tools
+        run: pip install pip-tools
```

Comment:
this should be version constrained in another requirements file... but so should build and hatchling and any other build deps: I'll handle that in another PR